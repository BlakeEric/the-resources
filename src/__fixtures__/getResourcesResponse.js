const getResourcesResponse = [
  {
    node: {
      id: 'd04da249-50e7-5b18-a992-d530e14831a7',
      frontmatter: {
        title: 'A List Apart',
        format: 'Article',
        focus: 'Front-end',
        skillLevel: ['Beginner','Moderate','Advanced']
      },
      fields: {
        slug: '/a-list-apart/'
      }
    }
  },
  {
    node: {
      id: '77a6be63-d3c7-50cb-bd29-438427560388',
      frontmatter: {
        title: 'Scrimba',
        format: 'Interactive',
        focus: 'Front-end',
        skillLevel: ['Beginner','Moderate']
      },
      fields: {
        slug: '/scrimba/'
      }
    }
  },
  {
    node: {
      id: '28feb8c5-acb2-5638-bdd6-b0457eff8db3',
      frontmatter: {
        title: 'Smashing Magazine',
        format: 'Article',
        focus: 'Multiple',
        skillLevel: ['Beginner','Moderate','Advanced']
      },
      fields: {
        slug: '/smashing-magazine/'
      }
    }
  },
  {
    node: {
      id: '744d8211-8c35-5c83-b8c1-ecb4ab48c0db',
      frontmatter: {
        title: 'The Odin Project',
        format: 'Tutorial',
        focus: 'Multiple',
        skillLevel: ['Beginner','Moderate']
      },
      fields: {
        slug: '/the-odin-project/'
      }
    }
  },
  {
    node: {
      id: '3b19247d-2ddd-558d-b0ce-14ec1972d52f',
      frontmatter: {
        title: 'Watch and Code',
        format: 'Video',
        focus: 'Front-end',
        skillLevel: ['Beginner']
      },
      fields: {
        slug: '/watch-and-code/'
      }
    }
  }
]

export default getResourcesResponse;
