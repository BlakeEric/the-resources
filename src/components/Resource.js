import React from 'react'
// import { Link } from "gatsby"
import { css } from '@emotion/core'
import { rhythm } from "../utils/typography"

export default function Resource(props) {

  const handleClick = (e) => {
    e.preventDefault();

    if (props.isToggled) {
      props.setToggledItemId(null)
    } else {
      props.setToggledItemId(props.id)
    }
  }


  return (
    <tr key={props.id}>
      <td>
        <a href="#resources" onClick={(e) => handleClick(e)}>
          {props.frontmatter.name}
        </a>
        {props.isToggled &&
          <div css={css`position: relative`}>
            <aside className="resourceDetails" css={resourceDetailStyles}>
              <h4>{props.frontmatter.name}</h4>
              <div dangerouslySetInnerHTML={{ __html: props.html }} />
              <a href={props.frontmatter.url}>View Now</a>
            </aside>
          </div>
        }
      </td>
      <td>
        {props.frontmatter.format}
      </td>
      <td>
        {props.frontmatter.focus}
      </td>
      <td>
        {props.frontmatter.skillLevel.length === 3 ? 'All' : props.frontmatter.skillLevel.join(', ')}
      </td>
    </tr>

  )
}

Resource.defaultProps = {
  isToggled: false,
  setToggledItemId: () => {}
}

const resourceDetailStyles = css`
  position: absolute;
  left: 0;
  max-width: 90vw;
  width: 300px;
  top: 5px;
  left: -${rhythm(.25)};
  background: white;
  border-radius: 5px;
  box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
  padding: ${rhythm(.25)};
  z-index: 10000;
  border: 1px solid #e4e4e4;
  &:after {
    content: "";
    height: 12px;
    width: 12px;
    position: absolute;
    top: 0;
    left: 0;
    background: white;
    transform: rotate(-45deg);
    transform-origin: 0 0;
    left: ${rhythm(1)};
    border-right: 1px solid #e4e4e4;
    border-top: 1px solid #e4e4e4;
  }
  h4 {
    margin-bottom: ${rhythm(.25)};
  }
  p {
    color: black;
  }
  p:last-child {
    margin: 0;
  }
`
