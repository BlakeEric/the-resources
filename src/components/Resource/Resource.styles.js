import { css } from '@emotion/core'
import { rhythm } from "../../utils/typography"

export const resourceDetailStyles = css`
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -${rhythm(.25)};
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: ${rhythm(.5)};
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: "";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: ${rhythm(.5)};
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: ${rhythm(.5)};
    }
    .btn-view {
      padding-right: ${rhythm(.25)};
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
`