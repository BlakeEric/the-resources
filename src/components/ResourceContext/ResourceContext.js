import React from 'react'


export const ResourceContext = React.createContext();


export class ResourceProvider extends React.Component {

  state = {
    bookmarks: [],
    currentToggledItemId: null,
    isFilterable: false, // used to display filter options if JS is enabled
    orderBy: "name",
    order: "ASC",
    resources: this.props.resources,
    searchTerm: null,
    selectedSkillLevel: null,
    selectedFormat: null,
    selectedFocus: null,
  }


  componentDidMount() {
    // if JS is enabled allow dynamic filtering
    this.setState({isFilterable: true})
  }


  /*
   * add an resource's id to bookmarks of saved resources
   */
  addToBookmarks = (id) => {
    this.setState({
      bookmarks: [...this.state.bookmarks, id]
    })
  }


  /*
   * remove an resource's id from bookmarks of saved resources
   */
  removeFromBookmarks = (id) => {
    this.setState({
      bookmarks: this.state.bookmarks.filter(item => item !== id)
    })
  }


  /*
   * Reverse value of 'order' in state: "ASC" or "DESC"
   */
  reverseOrder = () => {
    let currentOrder = this.state.order;
    let order;

    if (currentOrder === "DESC") {
      order = "ASC"
    } else {
      order = "DESC"
    }

    this.setState({
      order
    })
  }

  setToggledItemId = (id) => {
    this.setState({
      currentToggledItemId: id
    })
  }

  /*
   * Set the key to use as basis for alphabetical sorting
   */
  setOrderBy = (key) => {
    this.setState({
      orderBy: key,
      order: "ASC"
    })
  }


  /*
   * Click handler that reverses order in state if a new field is clicked
   * Otherwise sets orderBy value to new key
   */
  setOrder = (key) => {
    if (!this.state.isFilterable) {
      return;
    }

    if (this.state.orderBy === key) {
      this.reverseOrder();
    } else {
      this.setOrderBy(key);
    }
  }


  /*
   * Set search term in state
   */
  setSearchTerm = (searchTerm) => {
    this.setState({searchTerm})
  }


  /*
   * Set filters in state
   */
  setSelected = (key, value) => {
    this.setState({[key]: value})
  }


  /*
   * Clear all selected filters from state
   */
  clearFilters = () => {
    this.setState({
      searchTerm: null,
      selectedSkillLevel: null,
      selectedFormat: null,
      selectedFocus: null
    })
  }


  /*
   * Alphabetically compare one field against another
   * Return -1, 0, or 1 for use in Array.sort
   */
  compareFields(current, next) {

    if (this.state.order === "DESC") {

      if(current > next) { return -1; }
      else if(current < next) { return 1; }

    } else {

      if(current < next) { return -1; }
      else if(current > next) { return 1; }

    }

    return 0;
  }


  /*
   * Filter the resources based on orderBy and search values set in state
   * Return reordered array
   */
  filter = () => {

    const { searchTerm, selectedSkillLevel, selectedFocus, selectedFormat } = this.state;
    const key = this.state.orderBy || 'name';

    // create a new array of resources to avoid mutating the value in state
    let filteredItems = [...this.state.resources];

    // remove resources not matching the search term in state
    if (searchTerm) {
      let string = searchTerm.toLowerCase()

      filteredItems = filteredItems.filter(item => {
        return item.node.frontmatter.name.toLowerCase().includes(string)
      })
    }

    // remove resources not matching the skillLevel in state
    if (selectedSkillLevel) {
      let string = selectedSkillLevel.toLowerCase()

      filteredItems = filteredItems.filter(item => {
        return item.node.frontmatter.skillLevel.toString().toLowerCase().includes(string)
      })
    }

    // remove resources not matching the format in state
    if (selectedFormat) {
      let string = selectedFormat.toLowerCase()

      filteredItems = filteredItems.filter(item => {
        return item.node.frontmatter.format.toLowerCase().includes(string)
      })
    }

    // remove resources not matching the format in state
    if (selectedFocus) {
      let string = selectedFocus.toLowerCase()

      filteredItems = filteredItems.filter(item => {
        return item.node.frontmatter.focus.toLowerCase().includes(string)
      })
    }

    // Sort the items alphabetically by "orderBy" key set in state.
    // Will sort in ASC order by default, or DESC if set as "order" value in state
    filteredItems.sort((current, next) => {

      let currentField = current.node.frontmatter[key].toString().toLowerCase();
      let nextField = next.node.frontmatter[key].toString().toLowerCase();

      if (key === "skillLevel") {
        currentField = currentField.replace("beginner,moderate,advanced", "all");
        nextField = nextField.replace("beginner,moderate,advanced", "all");
      }

      return this.compareFields(currentField, nextField)

    })

    return filteredItems
  }


  render() {

    const contextValue = {
      ...this.state,
      reverseOrder: this.reverseOrder,
      setToggledItemId: this.setToggledItemId,
      setOrder: this.setOrder,
      setOrderBy: this.setOrderBy,
      getFilteredResources: this.filter,
      setSearchTerm: this.setSearchTerm,
      setSelected: this.setSelected,
      clearFilters: this.clearFilters,
      allFormats: this.props.allFormats,
      allFocuses: this.props.allFocuses,
      allSkillLevels: this.props.allSkillLevels,
      addToBookmarks: this.addToBookmarks,
      removeFromBookmarks: this.removeFromBookmarks
    }

    return (
      <ResourceContext.Provider value={contextValue}>
        {this.props.children}
      </ResourceContext.Provider>
    );
  }
}
