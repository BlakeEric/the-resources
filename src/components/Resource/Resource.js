import React, { useContext, useEffect } from 'react'
import { ResourceContext } from '../ResourceContext/ResourceContext'
import { Button } from 'theme-ui'
import { resourceDetailStyles } from "./Resource.styles"

export default function Resource(props) {

  const context = useContext(ResourceContext)

  /**
   * return true if the current item's details should be showing
   */
  const isToggled = context.currentToggledItemId === props.id


  /**
   * return true if the current item's details should be showing
   */
  const isInBookmarks = context.bookmarks.includes(props.id)

  /**
   * add/remove click handler to close detail window
   * when user clicks outside of the container
   */
  useEffect(() => {

    if(isToggled) {
      document.addEventListener('mousedown', handleClickOutside);
    } else {
      document.removeEventListener('mousedown', handleClickOutside);
    }

    // Specify how to clean up after this effect:
    return function cleanup() {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  });

  /**
   * Set the wrapper ref
   */
  let wrapperRef;
  const setWrapperRef = (node) => {
    wrapperRef = node;
  }

  /**
   * Set Trigger ref
   */
  let triggerRef;
  const setTriggerRef = (node) => {
    triggerRef = node;
  }

  /**
   * Close detail tab if clicked on outside
   */
  const handleClickOutside = (event) => {
    if (
      wrapperRef &&
      triggerRef &&
      !wrapperRef.contains(event.target) &&
      !triggerRef.contains(event.target)
    ) {
      context.setToggledItemId(null)
    }
  }

  /**
   * Open detail tab is currently closed, and vice versa
   */
  const handleToggle = (e) => {

    e.preventDefault();

    if (isToggled) {
      context.setToggledItemId(null)
    } else {
      context.setToggledItemId(props.id)
    }
  }


  return (
    <tr key={props.id}>
      <td>
        <div className="resourceDetails-wrapper" css={resourceDetailStyles}>
          <a
            href={props.frontmatter.url}
            onClick={(e) => handleToggle(e)}
            target="_blank" rel="noreferrer noopener"
            ref={setTriggerRef}
          >
            {props.frontmatter.name}
          </a>
          {isToggled &&
            <aside
              className={`resourceDetails ${props.showDetailsAbove ? 'up' : ''}`}
              ref={setWrapperRef}
            >
              <h4>{props.frontmatter.name}</h4>
              <div dangerouslySetInnerHTML={{ __html: props.html }} />
              <div className="controls">
                <a className="btn-view" href={props.frontmatter.url} target="_blank" rel="noreferrer noopener">View Now &rarr;</a>
                {isInBookmarks
                  ? <span className="right">Bookmarked! <Button variant="warning" onClick={() => context.removeFromBookmarks(props.id)}>Remove</Button></span>
                  : <Button variant="secondary" className="right" onClick={() => context.addToBookmarks(props.id)}>Add to bookmarks</Button>
                }
              </div>
            </aside>
          }
        </div>
      </td>
      <td>
        {props.frontmatter.format}
      </td>
      <td>
        {props.frontmatter.focus}
      </td>
      <td>
        {props.frontmatter.skillLevel.length === 3 ? 'All' : props.frontmatter.skillLevel.join(', ')}
      </td>
    </tr>
  )

}

Resource.defaultProps = {
  setToggledItemId: () => {}
}


