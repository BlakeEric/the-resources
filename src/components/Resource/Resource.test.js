import React from "react"
import renderer from "react-test-renderer"
import { ResourceContext } from "../ResourceContext/ResourceContext"
import getResourcesResponse from '../../__fixtures__/getResourcesResponse'

import Resource from "./Resource"


const mockProps = {
  ...getResourcesResponse[0].node,
  isToggled: false,
  showDetailsAbove: false
}

describe("Resource", () => {

  /*
  * Make sure the component renders without error
  */
  it("renders correctly not toggled", () => {
    const tree = renderer
      .create(
        <ResourceContext.Provider value={{setToggledItemId: () => {}}}>
          <Resource {...mockProps}/>)
        </ResourceContext.Provider>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly when toggled", () => {

    mockProps.isToggled = true;

    const tree = renderer
      .create(
        <ResourceContext.Provider value={{setToggledItemId: () => {}}}>
          <Resource {...mockProps}/>)
        </ResourceContext.Provider>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })


  it("renders popover above item when showDetailsAbove prop is true", () => {

    mockProps.isToggled = true;
    mockProps.showDetailsAbove = true

    const tree = renderer
      .create(
        <ResourceContext.Provider value={{setToggledItemId: () => {}}}>
          <Resource {...mockProps}/>)
        </ResourceContext.Provider>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

});
