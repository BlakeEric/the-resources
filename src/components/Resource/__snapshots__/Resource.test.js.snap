// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Shows indication that item has been bookmarked 1`] = `
Array [
  <tr>
    <td>
      <div
        className="resourceDetails-wrapper"
        css={
          Object {
            "map": undefined,
            "name": "1f9ea78",
            "next": undefined,
            "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
            "toString": [Function],
          }
        }
      >
        <a
          onClick={[Function]}
          rel="noreferrer noopener"
          target="_blank"
        >
          Alligator
        </a>
        <aside
          className="resourceDetails up"
        >
          <h4>
            Alligator
          </h4>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <div
            className="controls"
          >
            <a
              className="btn-view"
              rel="noreferrer noopener"
              target="_blank"
            >
              View Now →
            </a>
            <span
              className="right"
            >
              Bookmarked! 
              <button
                className="css-vhv7r1"
                onClick={[Function]}
              >
                Remove
              </button>
            </span>
          </div>
        </aside>
      </div>
    </td>
    <td>
      Article
    </td>
    <td>
      Front-end
    </td>
    <td>
      All
    </td>
  </tr>,
  ")",
]
`;

exports[`Resource renders correctly not toggled 1`] = `
Array [
  <tr>
    <td>
      <div
        className="resourceDetails-wrapper"
        css={
          Object {
            "map": undefined,
            "name": "1f9ea78",
            "next": undefined,
            "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
            "toString": [Function],
          }
        }
      >
        <a
          onClick={[Function]}
          rel="noreferrer noopener"
          target="_blank"
        >
          Alligator
        </a>
      </div>
    </td>
    <td>
      Article
    </td>
    <td>
      Front-end
    </td>
    <td>
      All
    </td>
  </tr>,
  ")",
]
`;

exports[`Resource renders correctly when toggled 1`] = `
Array [
  <tr>
    <td>
      <div
        className="resourceDetails-wrapper"
        css={
          Object {
            "map": undefined,
            "name": "1f9ea78",
            "next": undefined,
            "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
            "toString": [Function],
          }
        }
      >
        <a
          onClick={[Function]}
          rel="noreferrer noopener"
          target="_blank"
        >
          Alligator
        </a>
        <aside
          className="resourceDetails "
        >
          <h4>
            Alligator
          </h4>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <div
            className="controls"
          >
            <a
              className="btn-view"
              rel="noreferrer noopener"
              target="_blank"
            >
              View Now →
            </a>
            <button
              className="right css-vhv7r1"
              onClick={[Function]}
            >
              Add to bookmarks
            </button>
          </div>
        </aside>
      </div>
    </td>
    <td>
      Article
    </td>
    <td>
      Front-end
    </td>
    <td>
      All
    </td>
  </tr>,
  ")",
]
`;

exports[`Resource renders popover above item when showDetailsAbove prop is true 1`] = `
Array [
  <tr>
    <td>
      <div
        className="resourceDetails-wrapper"
        css={
          Object {
            "map": undefined,
            "name": "1f9ea78",
            "next": undefined,
            "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
            "toString": [Function],
          }
        }
      >
        <a
          onClick={[Function]}
          rel="noreferrer noopener"
          target="_blank"
        >
          Alligator
        </a>
        <aside
          className="resourceDetails up"
        >
          <h4>
            Alligator
          </h4>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <div
            className="controls"
          >
            <a
              className="btn-view"
              rel="noreferrer noopener"
              target="_blank"
            >
              View Now →
            </a>
            <button
              className="right css-vhv7r1"
              onClick={[Function]}
            >
              Add to bookmarks
            </button>
          </div>
        </aside>
      </div>
    </td>
    <td>
      Article
    </td>
    <td>
      Front-end
    </td>
    <td>
      All
    </td>
  </tr>,
  ")",
]
`;
